### PR TITLE
Optional Sentry introduction

### PR DESCRIPTION
**Pull request by magul**
It's going to publish sorucemap only when environment variables will be set, so it will happen only on Vercel project with Sentry integration enabled.

Also be aware that this is a PoC, at some point we could hide `withSetnry(toNextHandler())` in one function in `saleor-app-sdk` (it will probably require to set `@sentry/nextjs` as `pairDependency`, though).

### CODE REVIEW

**Review by witoszekdev — COMMENTED** [review]
Really nice, and simple to understand 🎉

One more thing I would add, would be some note that our project uses Sentry for logging by default and to an example in README on how to configure Sentry in Vercel.

**Review by zaiste — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]
I appreciate the comments - serverless bug tracking is not as easy.

Regarding moving it to the SDK: I'm not sold on the idea. Users of the SDK may use different monitoring solutions and it will additionally bloat the package. IMO having the example of the usage in this repository is enough

File: pages/_error.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+const MyError = ({
+  statusCode,
+  hasGetInitialPropsRun,
+  err,
+}: ErrorPageProps) => {
+  if (!hasGetInitialPropsRun && err) {
+    // getInitialProps is not called in case of
+    // https://github.com/vercel/next.js/issues/8592. As a workaround, we pass
```
[the remark this answers is not in the document]
Comment (magul) [reply pages/_error.tsx]: done, but I re-phrased you suggestion.

**Comment by magul** [comment]
> One more thing I would add, would be some note that our project uses Sentry for logging by default and to an example in README on how to configure Sentry in Vercel.

@krzysztofwolski promised to include that in next round of readme update.

**Comment by timuric** [comment]
sdk should be as lean as possible, and have very few dependencies. This is very important if it is being used in the storefront

**Pull request merged**